core/trino-main/src/main/java/io/trino/cost/TableScanStatsRule.java

```diff
@@ -105,4 +107,25 @@ else if (type instanceof FixedWidthType) {
         });
         return result.build();
     }
+
+    private static double getNullsFraction(ColumnStatistics columnStatistics, Estimate rowCount)
+    {
+        if (!columnStatistics.getNullsFraction().isUnknown()
+                || columnStatistics.getDistinctValuesCount().isUnknown()
+                || rowCount.isUnknown()) {
+            return columnStatistics.getNullsFraction().getValue();
+        }
+        // When NDV is greater than or equal to row count, there are no nulls
+        if (columnStatistics.getDistinctValuesCount().getValue() >= rowCount.getValue()) {
+            return 0;
+        }
+
+        double maxPossibleNulls = rowCount.getValue() - columnStatistics.getDistinctValuesCount().getValue();
+
+        // If a connector provides NDV but is missing nulls fraction statistic for a column
+        // (e.g. Delta Lake after "delta.dataSkippingNumIndexedCols" columns and MySql), populate a
+        // 10% guess value so that the CBO can still produce some estimates rather failing to make
+        // any estimates due to lack of nulls fraction.
+        return Math.min(UNKNOWN_NULLS_FRACTION, maxPossibleNulls / rowCount.getValue());
```

Comment:
what if `rowCount.getValue() == 0`? What is convention for null fraction there?
Add a test case?

Reply:
We return 0 nulls fraction for that due to the "ndv >= row count" check above. That gets converted into SymbolStatsEstimate.ZERO
Added a test